drivers/storage/portworx/component/csi.go

```diff
@@ -1167,6 +1196,12 @@ func (c *csi) getCSIConfiguration(
 			InstallSnapshotController: boolPtr(false),
 		}
 	}
+	if cluster.Spec.CSI.Enabled && cluster.Spec.CSI.Topology == nil {
+		// CSI topology feature gate is disabled by default
+		cluster.Spec.CSI.Topology = &corev1.CSITopologySpec{
```

Comment:
@piyush-nimbalkar @ggriffiths do you think we should enable it by default?

Reply:
as it's only for direct access I think it's better to leave it disabled by default. also here the cluster object is just used for populating the csi configuration I don't think the actual object will be updated, as by default the whole topology field should be nil.